evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -754,7 +772,49 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> Handler for CheatcodeStackExecutor<'a
                 true,
                 true,
                 context,
-            )
+            );
+
+            if let Some(expected_revert) = expected_revert {
+                let final_res = match res {
+                    Capture::Exit((ExitReason::Revert(_e), data)) => {
+                        if data.len() >= 4 && data[0..4] == [8, 195, 121, 160] {
+                            // its a revert string
+                            let len = U256::from(&data[4 + 32..36 + 32]).as_usize();
+                            if data[4 + 32 + 32..4 + 32 + 32 + len] == *expected_revert {
+                                return Capture::Exit((
+                                    ExitReason::Succeed(ExitSucceed::Returned),
+                                    DUMMY_OUTPUT.to_vec(),
+                                ))
+                            } else {
+                                return evm_error(&*format!(
+                                    "Error != expected error: '{}' != '{}'",
+                                    String::from_utf8_lossy(
+                                        &data[4 + 32 + 32..4 + 32 + 32 + len]
+                                    ),
+                                    String::from_utf8_lossy(expected_revert)
+                                ))
+                            }
+                        }
+
+                        if data == *expected_revert {
```

Comment:
Isn't this logic duplicated above? Should we not just error if the condition above (`data.len() ...`) is not hit?

Reply:
no - the first code path is a revert string test, the second is for custom errors